### DISCUSSION TITLE
vcpkg manifest mode doesn't behave nicely for non-root user in linux

### DISCUSSION THREAD

**Discussion by ericfont** [microsoft/vcpkg#20541] [Q&A] [answered]
I was trying to learn how to use manifest mode and following the example (https://github.com/microsoft/vcpkg/blob/master/docs/users/manifests.md) which I've copied exactly into this git repo: https://github.com/ericfont/fibo-vcpkg-example

I am on arch linux x86-64 and installed [vcpkg-git 2021.05.12.r575.g6b24f0fa76-1 via the Arch User Repository](https://aur.archlinux.org/packages/vcpkg-git/) which installs to `/usr/share/vcpkg/`, and I can easily install and uninstall packages using `sudo vcpkg`, which is great.  And there is a root lockfile `/usr/share/vcpkg/.vcpkg-root` which I presume is to lock the install database when a user with sudo or root privileges is installing packages.  And there is a directory `/root/.cache/vcpkg/archives/` containing a bunch of zip files names with long hashes which I believe correspond to packages I've installed in the past which might be reused...

However while that works fine for running `sudo vcpkg install` as a regular user, it led to problems when I was trying to build the manifest example.  Running the command:

`cmake -B build -S . -DCMAKE_TOOLCHAIN_FILE=/usr/share/vcpkg/scripts/buildsystems/vcpkg.cmake`

inside of the fibonacci example project folder without `sudo` permissions resulted in a bunch of CMAKE errors you can see here: https://github.com/ericfont/fibo-vcpkg-example/issues/1

I found a workaround hack to prepend that `cmake` command with `sudo`, but the problem with that it is not good practice to run commands with root privileges (such as cmake) unless it is really absolutely necessary (and it shouldn't really be necessary with manifest mode because the packages are just being installed inside the local project folder, not installed in the system's root folder), and because I don't like having my local project's build folder and files under root ownership (cause then for instance I have to use sudo permissions when removing the build folder, which is dangerous if I have a fat finger error).

What would be preferred is a way to separate any necessary package installs to the root vcpkg to be run as a separate command from the cmake command, that way I can use `sudo` only for installing the vcpkgs to my system's folder but then be able to run the cmake command as regular user.  Or alternatively maybe have a vcpkg unix group and any users in that group can install vcpkg packages inside /usr/share/vcpkg and that way I don't have to escalate my user to sudo privileges just to install vcpkgs.

**Comment by autoantwort** [comment]
It is discouraged to install vcpkg via a package manager. Use `git clone` to install it, then you also don't have this problem.

**Comment by dg0yt** [comment, marked as answer]
I don't think it is a good idea to run `sudo vcpkg` at all, unless you are on a throw-away system.

**Comment by ericfont** [comment]
OK, thanks.  I guess I should tell that Arch User Repo package maintainer to delete it since it promotes bad practice.

This reminds me of why people say to avoid `sudo pip install`

**Comment by ericfont** [comment]
Thanks for your answers...I've forwarded them to the AUR package: https://aur.archlinux.org/packages/vcpkg-git/#comment-830069
